### PR TITLE
#633. @Builder.Default causes a warning message

### PR DESCRIPTION
**Pull request by Lekanich**
- @Builder.Default doesn't causes a warning message now

### CODE REVIEW

**Comment by mplushnikov** [comment]
Thank you again @Lekanich ! Will review it the next days... Maybe you can rebase it in the time:-)

File: src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderInfo.java [hunk 1]
```diff
@@ -24,8 +24,8 @@
   private static final String BUILDER_OBTAIN_VIA_FIELD = "field";
   private static final String BUILDER_OBTAIN_VIA_METHOD = "method";
   private static final String BUILDER_OBTAIN_VIA_STATIC = "isStatic";
-  private static final String BUILDER_OBTAIN_VIA_ANNOTATION = Builder.ObtainVia.class.getName().replace("$", ".");
-  private static final String BUILDER_DEFAULT_ANNOTATION = Builder.Default.class.getName().replace("$", ".");
+  private static final String BUILDER_OBTAIN_VIA_ANNOTATION = Builder.ObtainVia.class.getCanonicalName();
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderInfo.java]: Excellent! Feel free to update all places!

File: src/main/java/de/plushnikov/intellij/plugin/handler/BuilderHandler.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import com.intellij.psi.PsiField;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.PsiTreeUtil;
+import lombok.Builder;
+import de.plushnikov.intellij.plugin.util.PsiAnnotationSearchUtil;
+
+
+public class BuilderHandler {
+
+  public static boolean isDefaultBuilderValue(HighlightInfo highlightInfo, PsiFile file) {
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/handler/BuilderHandler.java]: You can add here @NotNull annotations too.

File: src/main/java/de/plushnikov/intellij/plugin/extension/LombokHighlightErrorFilter.java [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
     if (null != file) {
-
+      String description = StringUtil.notNullize(highlightInfo.getDescription());
       if (HighlightSeverity.WARNING.equals(highlightInfo.getSeverity())) {
-        if (REDUNDANT_DEFAULT_PARAMETER_VALUE_ASSIGNMENT.equals(highlightInfo.getDescription())) {
+        if (REDUNDANT_DEFAULT_PARAMETER_VALUE_ASSIGNMENT.equals(description)) {
           return !EqualsAndHashCodeCallSuperHandler.isEqualsAndHashCodeCallSuperDefault(highlightInfo, file);
+        } else if (CodeInsightColors.NOT_USED_ELEMENT_ATTRIBUTES.equals(highlightInfo.type.getAttributesKey())) {
+          if (VARIABLE_INITIALIZER_IS_REDUNDANT.matcher(description).matches()) {
+            return !BuilderHandler.isDefaultBuilderValue(highlightInfo, file);
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/extension/LombokHighlightErrorFilter.java]: You can try to write an inspection test, I think this warning is coming from DefUseInspection. Example: src/test/java/de/plushnikov/intellij/plugin/inspection/DataFlowInspectionTest.java

Comment (Lekanich) [reply src/main/java/de/plushnikov/intellij/plugin/extension/LombokHighlightErrorFilter.java]: Thanks for the example and review. I'll plane to cover it with a test. Also, I'm thinking about a small refactoring in this class. I'll push it in current PR.

File: src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderInfo.java (hunk 1 above)
Comment (Lekanich) [reply src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderInfo.java]: Ok)

File: src/main/java/de/plushnikov/intellij/plugin/handler/BuilderHandler.java (hunk 2 above)
Comment (Lekanich) [reply src/main/java/de/plushnikov/intellij/plugin/handler/BuilderHandler.java]: Yes, sure

**Comment by Lekanich** [comment]
@mplushnikov Please take a look

**Comment by mplushnikov** [comment]
Thank you @Lekanich ! Just merged it. I only changed registration of the filters a little bit. (https://github.com/mplushnikov/lombok-intellij-plugin/commit/d60d3a7e900f02e72ffd637f4dc858d3a42330d0)

**Pull request merged**